### ISSUE TITLE
Create documentation for how to run our site with jekyll

### ISSUE THREAD

**Issue by jbubar** [hackforla/website#1674]
### Overview
Sometimes docker does not work, so we should have some documentation about how to use jekyll.

### Action Items
- [ ] We are currently in the process of moving the old wiki to the new [website-wiki repo](https://github.com/hackforla/website-wiki), so we will not be making any changes or additions to the old wiki at this time. Thus, we will be adding wiki content through a different process now. Please read [How to Contribute to the Wiki](https://github.com/hackforla/website/wiki/How-to-Contribute-to-the-Wiki)
- [ ] create a wiki article with the necessary steps
- [ ] test it out
- [ ] Make sure it is compatible with all systems (Mac, Windows, Iinux)
- [ ] share with dev team and have people review it
- [ ] close this issue

### Plan
- Create a Gemfile.example (Gemfile is in .gitignore.  See comments below for discussions about Gemfile and why it can't be checked in) 
- create separate markdown on deploying in jekyll. Indicate in the document that if you run into issues, support may be limited and you may be asked to try Docker.
- Link to this file from contributing.md with same caveat as above.

### Instruction
- downloading [ruby](https://rubyinstaller.org/) and [jekyll](https://jekyllrb.com/docs/installation/) for windows
- other instructions for mac
- cd into directory w website repo
- cp Gemfile.example Gemfile
- type `bundle install`
- type `jekyll serve`

**Comment by jbubar** [comment]
@cnk @akibrhast do we need a Gemfile? and also, are there any flags to add to the `jekyll serve` command that would be useful?

**Comment by cnk** [comment]
To run Jekyll without Docker, one would need a Gemfile. The problem is, if there is a Gemfile in the repository, it interferers with the built in Gemfile in the Docker container. 

Since the important thing is that our code match what GitHub is going to do with our files once pushed, I have been reluctant to do anything other than use the prebuilt, already working, Docker container. If you still want to be able to set up the HackForLA website project not to need Docker, I recommend storing a `Gemfile.example` in the repo - or in the wiki. That should hide it from GitHub pages. AND make sure you match supported versions of Jekyll. GH pages does not run the latest version of Jekyll - 3.9.0 NOT Jekyll 4. https://pages.github.com/versions/

**Comment by Sihemgourou** [comment]
@jbubar please add a role label to this and remove the role missing label. Thanks !

**Comment by adam-abundis** [comment]
@jbubar What would you like to discuss about this issue?

**Comment by AzaniaBG** [comment]
ETA: 09/28/2022 by 5 PM
Availability: 09/26/2022 3 - 4 PM; 09/27/2022 2 - 4 PM; 09/28/2022 2 - 4 PM

**Comment by ExperimentsInHonesty** [comment]
This was on an agenda 2022-10-02

> Azania Baker-Garcia's issues
https://github.com/hackforla/website/issues/1674
do we want a note about prioritizing using docker/discouraging people from using Jekyll
Bonnie:
Ethan had an issue with Docker (took a long time), figured out how to run Jeykll
without docker -not a high priority until it is
We should talk to Ethan about how to walk through it.
https://github.com/hackforla/website/issues/2607#issuecomment-1264726733

It then got moved to the icebox.  But it has no dependency, so cannot stay in the icebox.  Once you add a dependency to the top of the issue and a dependency label, it can go back into the icebox.

**Comment by AzaniaBG** [comment]
> 

I think website leads (@kathrynsilvaconway, @blulady) were trying to figure out whether it should be an issue at all...

**Comment by blulady** [comment]
> This was on an agenda 2022-10-02
> 
> > Azania Baker-Garcia's issues
> > #1674
> > do we want a note about prioritizing using docker/discouraging people from using Jekyll
> > Bonnie:
> > Ethan had an issue with Docker (took a long time), figured out how to run Jeykll
> > without docker -not a high priority until it is
> > We should talk to Ethan about how to walk through it.
> > [#2607 (comment)](https://github.com/hackforla/website/issues/2607#issuecomment-1264726733)
> 
> It then got moved to the icebox. But it has no dependency, so cannot stay in the icebox. Once you add a dependency to the top of the issue and a dependency label, it can go back into the icebox.

I reached out to @ethanstrominger and he told me I was mistaken, that he didn't have any issues with docker. Is there another Ethan who might have had this issue?

**Comment by AzaniaBG** [comment]
> > This was on an agenda 2022-10-02
> > > Azania Baker-Garcia's issues
> > > #1674
> > > do we want a note about prioritizing using docker/discouraging people from using Jekyll
> > > Bonnie:
> > > Ethan had an issue with Docker (took a long time), figured out how to run Jeykll
> > > without docker -not a high priority until it is
> > > We should talk to Ethan about how to walk through it.
> > > [#2607 (comment)](https://github.com/hackforla/website/issues/2607#issuecomment-1264726733)
> > 
> > 
> > It then got moved to the icebox. But it has no dependency, so cannot stay in the icebox. Once you add a dependency to the top of the issue and a dependency label, it can go back into the icebox.
> 
> I reached out to @ethanstrominger and he told me I was mistaken, that he didn't have any issues with docker. Is there another Ethan who might have had this issue?

That is the same Ethan I spoke with @blulady. I was under the same impression, but perhaps it was just easier for him to use Jekyll?

**Comment by ExperimentsInHonesty** [comment]
I think it's worth making if someone want to, and we are short of issues that are frontend OR backend AND medium.  It's possible no one will ever need it, but that's why it has the milestone it does.

Most problems that this solution would address are already addressed in https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#i-what-if-you-cannot-see-your-changes-locally-within-docker

But if you are reading this and want to work on it, jump on it!

**Comment by blulady** [comment]
@ExperimentsInHonesty 
Ethan Strominger
  [1:50 PM](https://hackforla.slack.com/archives/C04C1AV1M33/p1668808203138679)
Aziana - See thread below, followed by a summary
Hey Ethan, do you think that your experience would allow you to complete this [issue](https://github.com/hackforla/website/issues/1674)?
Ethan Strominger  [7:12 AM]
Yes.  However, I had volunteered to document Jekyll earlier and Aziana, Bonnie, and others suggested documenting how to use Jekyll was not a good idea.  Docker is an adequate solution and documenting how to use Jekyll could result in devs asking for support in doing this when they run into a problem - better if they have to support only one method of deploying and Docker is more foolproof. (edited)
[7:14 AM] So, maybe close this as not an issue?
[7:15 AM] I could add a note into the documentation that you can try to run in Jekyll without any instructions but you are on your own if you do - this way someone won’t suggest the issue in the future.
[7:19 AM] I just saw Bonnie’s last comment that it could be worth it - I would want to check with Aziana and Bonnie again - in the meeting I had with Bonnie, she agreed with the above argument for not documenting it. (edited)

**Comment by ethanstrominger** [comment]
Plan:

- Create a separate markdown on deploying in jekyll.  Indicate in the document that if you run into issues, support may be limited and you may be asked to try Docker.

- Link to this file from contributing.md with same caveat as above.

**Comment by ExperimentsInHonesty** [comment]
@ethanstrominger If you are still working on this, it should be in the in progress column.  If you are working on another issue and plan to pick this one up later, please let us know if it has any dependencies (like a different issue HAS to be completed before this can be worked on).

**Comment by JessicaLucindaCheng** [comment]
@ethanstrominger 
We are currently in the process of moving the old wiki to the new [website-wiki repo](https://github.com/hackforla/website-wiki), so we will not be making any changes or additions to the old wiki at this time. Thus, we will be adding wiki content through a different process now. Please read [How to Contribute to the Wiki](https://github.com/hackforla/website/wiki/How-to-Contribute-to-the-Wiki)
